src/main/java/com/faforever/server/map/MapService.java

```diff
@@ -31,6 +38,22 @@ public MapVersion getRandomLadderMap() {
     return ladder1v1Maps.get(random.nextInt(ladder1v1Maps.size() - 1));
   }
 
+  public void incrementTimesPlayed(Map map) {
```

Comment:
Oh I forgot this one: Needs `@Transactional` as well :)

calling `getMapStats` from here will *not* trigger the transaction because it's a call from the object to the obejct itself - it will therefore not go through the Spring Bean Proxy so there is no way for Spring to intercept the call.